src/main/java/com/direwolf20/mininggadgets/common/items/MiningGadget.java

```diff
@@ -97,14 +99,17 @@ public void addInformation(ItemStack stack, @Nullable World world, List<ITextCom
         super.addInformation(stack, world, tooltip, flag);
 
         List<Upgrade> upgrades = UpgradeTools.getUpgrades(stack);
-        if(!Screen.hasShiftDown())
+        Minecraft mc = Minecraft.getInstance();
+
+        if(!InputMappings.isKeyDown(mc.mainWindow.getHandle(), mc.gameSettings.keyBindSneak.getKey().getKeyCode()))
```

Comment:
Screen.hasShiftDown worked fine here :P How come you changed it to InputMappings?

Reply:
InputMapping is used to get the key assigned to crouch instead of being heavy coded to shift

hasShiftDown() is a method that runs InputMapping but is strictly coded to return true on either left or right shift

Reply:
`mc.gameSettings.keyBindSneak.isPressed()` ?

Reply:
isPressed would only fire once and I had issues with isDown in where it'd be true for a tick and return false even though key was down. Some wonky stuff lol...I'd assume it would be easy as that  but NOPE